pyedflib/tests/test_edfreader.py

```diff
@@ -35,7 +38,14 @@ def tearDownClass(cls):
                 os.remove(os.path.join(data_dir, file))
             except Exception as e:
                 print(e)
-
+                
+    def tearDown(self):
+        # small hack to close handles in case of tests throwing an exception
+        for obj in gc.get_objects():
+            if isinstance(obj, (EdfWriter, EdfReader)):
+                obj.close()
+                del obj
+                
```

Comment:
if a test fails it will leave the file open. if the same file is accessed in a later test, that will fail again, just due to the file still being open.

added this to close all files that were not closed because a test failed.